base/runtime/core_builtin_soa.odin

```diff
@@ -425,4 +425,101 @@ clear_soa_dynamic_array :: proc(array: ^$T/#soa[dynamic]$E) {
 @builtin
 clear_soa :: proc{
 	clear_soa_dynamic_array,
-}
\ No newline at end of file
+}
+
+// Converts soa slice into a soa dynamic array without cloning or allocating memory
+@(builtin, require_results)
+into_dynamic_soa :: proc(array: $T/#soa[]$E) -> #soa[dynamic]E {
+	d: #soa[dynamic]E
+	footer := raw_soa_footer_dynamic_array(&d)
+	footer^ = {
+		cap = len(array),
+		len = 0,
+		allocator = nil_allocator(),
+	}
+
+	field_count: uintptr
+	when intrinsics.type_is_array(E) {
+		field_count = len(E)
+	} else {
+		field_count = uintptr(intrinsics.type_struct_field_count(E))
+	}
+
+	array := array
+	dynamic_data := uintptr(&d)
+	slice_data := uintptr(&array)
+	for _ in 0..<field_count {
+		(^uintptr)(dynamic_data)^ = (^uintptr)(slice_data)^
+		dynamic_data += size_of(rawptr)
+		slice_data += size_of(rawptr)
+	}
+
+	return d
+}
```

Comment:
I don't think should be `builtin` whatsoever. The equivalent isn't for "aos" stuff isn't available as being `builtin`.

Reply:
That's a good point. Do you think it should still be in the runtime package then, or would it be better suited for "core:slice"?

Reply:
Maybe a different package. I am not sure yet.

Reply:
Okay, I'll keep it in runtime for now. In the future I would support a "core:soa". I actually was thinking about trying to contribute a sort_soa sometime. Combined with soa_zip it might be a pretty good alternative to using sort_with_indices.